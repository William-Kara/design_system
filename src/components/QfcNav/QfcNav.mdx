import { ArgTypes, Meta } from '@storybook/blocks';
import QfcNav from '@/components/QfcNav/QfcNav.vue';

<Meta title="Elements/Nav" />

# Definition

QfcNav Component provides current position in the application and simple navigation.

## Props

The following props are available for the QfcNav component:

<ArgTypes of={QfcNav} />

# Usage

The navItems array prop is required. Each object should contains a text ("label") and a link ("to").

To use QfcNav, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/bulma/components/navbar.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcNav.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcNavItem.scss';

```

```html
<template>
  <QfcNav :navItems="navItems" />
</template>
```
