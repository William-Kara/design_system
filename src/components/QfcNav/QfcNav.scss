@include desktop {
  .navbar-center {
    display: flex;
    align-content: center;
    justify-content: center;
    width: 100%;
  }
}
@include touch {
  .navbar-center {
    .navbar-item {
      margin-bottom: 1rem;
    }
  }
}