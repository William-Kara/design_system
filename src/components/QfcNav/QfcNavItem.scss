
.navbar-item {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  margin-right: 3rem;
  color: $text;
  &:hover {
    border-radius: $br-m;
    color: $text;
  }
  &:focus {
    background-color: unset;
    color: inherit;
  }
  &.router-link-exact-active {
    color: $primary;
    font-weight: $semibold;
    &:hover {
      color: $primary;
      background-color: transparent;
    }
  }
}
@include desktop {
  .navbar-item {
    flex-grow: initial;
    flex-shrink: initial;
  }
}
