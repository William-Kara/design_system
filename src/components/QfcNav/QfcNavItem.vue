<script lang="ts" setup>
type Props = {
  label: string;
  to: string;
  isExternal?: Boolean;
};
defineProps<Props>();
const emit = defineEmits(['on-click']);
const onClick = () => {
  emit('on-click');
};
</script>

<template>
  <a v-if="isExternal" class="navbar-item" :href="to" @click="onClick" target="_blank">
    {{ label }}
  </a>
  <router-link v-else class="navbar-item" :to="to" @click="onClick">
    {{ label }}
  </router-link>
</template>
<style lang="scss" scoped src="./QfcNavItem.scss" />
