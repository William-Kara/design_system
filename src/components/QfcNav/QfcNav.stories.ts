import QfcNav from './QfcNav.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof QfcNav> = {
  component: QfcNav,
  title: 'Elements/Nav',
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    centerNavItems: [
      { label: 'Title 1', to: '/' },
      { label: 'Title 2', to: '/' },
      { label: 'Title 3', to: '/' },
      { label: 'Title 4', to: '/' },
    ],
  },
};
