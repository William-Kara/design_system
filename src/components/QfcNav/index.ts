import QfcNav from './QfcNav.vue';

export { QfcNav };
export default QfcNav;
