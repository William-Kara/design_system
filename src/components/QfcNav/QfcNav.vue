<script lang="ts" setup>
import QfcNavItem from './QfcNavItem.vue';

interface NavItems {
  label: string;
  to: string;
}
interface Props {
  navItems: NavItems[];
}
withDefaults(defineProps<Props>(), {});
</script>
<template>
  <div class="navbar-center">
    <QfcNavItem
      v-for="({ label, to, isExternal }, index) in navItems"
      :key="index"
      :label="label"
      :to="to"
      :isExternal="isExternal"
    >
    </QfcNavItem>
  </div>
</template>
<style lang="scss" scoped src="./QfcNav.scss" />
