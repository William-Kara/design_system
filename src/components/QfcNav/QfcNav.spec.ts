import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcNav from './QfcNav.vue';

let wrapper: any = null;

const propsValues = {
  navItems: [
    { label: 'Title 1', to: '' },
    { label: 'Title 2', to: '' },
    { label: 'Title 3', to: '' },
    { label: 'Title 4', to: '' },
  ],
};

beforeEach(() => {
  wrapper = shallowMount(QfcNav, {
    propsData: propsValues,
  });
});

describe('QfcNav Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
