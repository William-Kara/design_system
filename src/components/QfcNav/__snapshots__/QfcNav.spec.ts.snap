// Vitest Snapshot v1

exports[`QfcNav Component > should render correctly 1`] = `
"<div data-v-936bb0b7=\\"\\" class=\\"navbar-center\\">
  <qfc-nav-item-stub data-v-936bb0b7=\\"\\" label=\\"Title 1\\" to=\\"\\"></qfc-nav-item-stub>
  <qfc-nav-item-stub data-v-936bb0b7=\\"\\" label=\\"Title 2\\" to=\\"\\"></qfc-nav-item-stub>
  <qfc-nav-item-stub data-v-936bb0b7=\\"\\" label=\\"Title 3\\" to=\\"\\"></qfc-nav-item-stub>
  <qfc-nav-item-stub data-v-936bb0b7=\\"\\" label=\\"Title 4\\" to=\\"\\"></qfc-nav-item-stub>
</div>"
`;
